client/lib/analytics/ad-tracking/woo/record-gtm-datalayer-event.ts

```diff
@@ -0,0 +1,38 @@
+import { mayWeTrackByTracker } from '../../tracker-buckets';
+import { debug, TRACKING_IDS } from '../constants';
+import { initGTMContainer, loadGTMContainer } from '../gtm-container';
+
+/**
+ * Sends a step event to Google Tag Manager for the woo express stepper flow.
+ *
+ * @param {string} stepName
+ * @returns {void}
+ */
+export default async function recordGTMDatalayerEvent( stepName = '' ) {
+	if ( stepName ) {
+		loadGTMContainer( TRACKING_IDS.wooGoogleTagManagerId )
+			.then( () => initGTMContainer() )
+			.then( () => {
+				debug(
+					`recordGTMDatalayerEvent: Initialized GTM container ${ TRACKING_IDS.wooGoogleTagManagerId }`
+				);
+
+				// We ensure that we can track with GTM
+				if ( ! mayWeTrackByTracker( 'googleTagManager' ) ) {
```

Comment:
Can we check consent higher up in this function, i.e. before loading and initializing GTM? I think it should work.

Reply:
Makes sense; better to return early if tracking is not permitted. Updated the PR.